File: app/src/main/java/com/example/bmi_calc/BMI_Results.java
package com.example.bmi_calc;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.jjoe64.graphview.GraphView;
import com.jjoe64.graphview.series.DataPoint;
import com.jjoe64.graphview.series.LineGraphSeries;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class BMI_Results extends AppCompatActivity {

    GraphView graphView;

    String filename = "bmi.txt";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmi_results);

        graphView = findViewById(R.id.idGraphView);

        File file = new File(getFilesDir(), filename);

        Scanner scanner = null;
        try {
            scanner = new Scanner(file);
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        }
        ArrayList<Double> values = new ArrayList<>();

        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            String[] parts = line.split("\n");
            for (String part : parts) {
                double value = Double.parseDouble(part.trim());
                values.add(value);
            }
        }


        DataPoint[] dataPoints = new DataPoint[values.size()];
        for (int i = 0; i < values.size(); i++) {
            dataPoints[i] = new DataPoint(i, values.get(i));
        }

        LineGraphSeries<DataPoint> series = new LineGraphSeries<DataPoint>(dataPoints);

        graphView.setTitle(getString(R.string.button_BMI_results));


        graphView.setTitleColor(R.color.purple_700);


        graphView.setTitleTextSize(45);


        graphView.addSeries(series);
        graphView.getViewport().setScrollable(true);
        graphView.getViewport().setScalable(true);


    }
}

File: app/src/main/java/com/example/bmi_calc/BMI_Results.kt
package com.example.bmi_calc

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.jjoe64.graphview.GraphView
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries
import java.io.File
import java.io.FileNotFoundException
import java.util.*

class BMI_Results : AppCompatActivity() {
    lateinit var graphView: GraphView
    var filename = "bmi.txt"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmi_results)
        graphView = findViewById(R.id.idGraphView)
        val file = File(filesDir, filename)
        val scanner: Scanner
        scanner = try {
            Scanner(file)
        } catch (e: FileNotFoundException) {
            throw RuntimeException(e)
        }
        val values = ArrayList<Double>()
        while (scanner.hasNextLine()) {
            val line = scanner.nextLine()
            val parts = line.split("\n".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            for (part in parts) {
                val value = part.trim { it <= ' ' }.toDouble()
                values.add(value)
            }
        }
        val dataPoints = arrayOfNulls<DataPoint>(values.size)
        for (i in values.indices) {
            dataPoints[i] = DataPoint(i.toDouble(), values[i])
        }
        val series = LineGraphSeries(dataPoints)
        graphView.title = getString(R.string.button_BMI_results)
        graphView.titleColor = R.color.purple_700
        graphView.titleTextSize = 45f
        graphView.addSeries(series)
        graphView.viewport.isScrollable = true
        graphView.viewport.isScalable = true
    }
}